### ISSUE TITLE
bug: 20240530212419_model_price_anthropic_via_google_vertex fails

### ISSUE THREAD

**Issue by patrit** [langfuse/langfuse#2266]
### Describe the bug

The migration 20240530212419_model_price_anthropic_via_google_vertex fails:

```
Prisma schema loaded from packages/shared/prisma/schema.prisma
Datasource "db": PostgreSQL database "dummy", schema "langfuse" at "db:5432"

201 migrations found in prisma/migrations

Error: P3009

migrate found failed migrations in the target database, new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
The `20240530212419_model_price_anthropic_via_google_vertex` migration started at 2024-06-07 05:59:00.783912 UTC failed
```

Log in the _prisma_migrations:
```
A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve

Migration name: 20240530212419_model_price_anthropic_via_google_vertex

Database error code: 42P01

Database error:
ERROR: relation "public.models" does not exist

Position:
[1m  0[0m
[1m  1[0m -- Google Vertex uses @ to separate model name and version
[1m  2[0m
[1m  3[1;31m UPDATE "public"."models" SET "match_pattern" = '(?i)^(claude-3-haiku(-|@)?20240307)$' WHERE "id" = 'cltr0w45b000008k1407o9qv1';[0m

DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E42P01), message: "relation \"public.models\" does not exist", detail: None, hint: None, position: Some(Original(68)), where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("parse_relation.c"), line: Some(1371), routine: Some("parserOpenTable") }

   0: sql_schema_connector::apply_migration::apply_script
           with migration_name="20240530212419_model_price_anthropic_via_google_vertex"
             at schema-engine/connectors/sql-schema-connector/src/apply_migration.rs:106
   1: schema_core::commands::apply_migrations::Applying migration
           with migration_name="20240530212419_model_price_anthropic_via_google_vertex"
             at schema-engine/core/src/commands/apply_migrations.rs:91
   2: schema_core::state::ApplyMigrations
             at schema-engine/core/src/state.rs:226
```

### To reproduce

Using official langfuse 2.48.0 container with a connection string using another schema
i.e. non default public
e.g. DATABASE_URL=postgresql://postgres:postgres@db:5432/dummy?schema=langfuse

### Additional information

IMHO one started to pin the supported db schema to "public" while not required previously. This affects db migrations starting with 20240530212419_model_price_anthropic_via_google_vertex. 20240606090858_pricings_add_latest_gemini_models also seems to be affected. Removing the "public". in these scripts makes them execute in the desired schema. This is only a bug if schema != "public" is supported, which I hope is the case.

### Are you interested to contribute a fix for this bug?

No

**Comment by marcklingen** [comment]
Thanks for reporting this! We are investigating potential fixes to edit these migrations without breaking deployments.

You are right, this is an unintended change which was not caught in CI as these all run un the public schema.

**Comment by maxdeichmann** [comment]
Hi @patrit thanks a lot for raising this. You were right, we started using the public schema which we never did before. As the migrations did not contain a schema change, but only added data to the models table, we were able to alter migration history in this [PR](https://github.com/langfuse/langfuse/pull/2329/files).
You can deploy the new container and it will clean up your unsuccessful state and apply everything correctly.

With this change we also ensured that you can use Langfuse with Postgres >=12 and on any postgres schema.

We released this fix in [v2.54.0](https://github.com/langfuse/langfuse/releases/tag/v2.54.0).
